### ISSUE TITLE
How to set offsetParent for a child component?

### ISSUE THREAD

**Issue by MoNSTRiKcom** [bcakmakoglu/revue-draggable#166]
Not working example:

`:offsetParent="$parent.$el"`
`:offsetParent="$parent.$refs.parentTable"`
`offsetParent="div.parentTable"`

p.s. It is necessary to expand the documentation with examples for each parameter with non-default values. I'll torture you :)

**Comment by bcakmakoglu** [comment]
What exactly are you trying to accomplish by passing the offset parent?
If you're trying to keep an element "in-bounds" use the bounds option and pass an element selector (i.e. an id, class name etc.).

The offset parent option does work though but not for keeping an element inside boundaries, which is what I am guessing you're trying to do.

I will try to extend on the docs, I just got a lot of work going on so don't hesitate to open an issue and/or ask questions.

**Comment by MoNSTRiKcom** [comment]
Yes, you must specify the boundaries of the displacement field.

The DOM structure is like this:

".parentTable // parent component
____> .container // child component
________> Draggable"

I need to specify a different parent and not an intermediary. I thought for this there is offsetParent, but it cannot be reassigned.

**Comment by bcakmakoglu** [comment]
```
<Draggable bounds=".parentTable">...</Draggable>
```

Does this not solve your issue then? You want the elements to be contained inside of the parentTable, right?

**Comment by MoNSTRiKcom** [comment]
It worked, but it doesn't work well. From the current position does not go higher and to the left. And the races became strange.

**Comment by bcakmakoglu** [comment]
What do you mean exactly? Can you give me an example or idea what the problem is? I can't picture anything regarding races or what you mean by going higher and to the left 😅

**Comment by MoNSTRiKcom** [comment]
Sorry. Didn't check the translation of the text))))
The block stopped moving to the left and up. Weird grid jumps have appeared.

**Comment by bcakmakoglu** [comment]
Are you certain it's not moving in that direction because the elements boundaries are actually at that position?
What do the dev-tools say, can you show me a screenshot or something as a reference?

**Comment by MoNSTRiKcom** [comment]
[Video link](https://drive.google.com/file/d/1JzHnkCcD8KSFN1HQK8qYA9aGIbOhgzpi/view?usp=sharing)

**Comment by bcakmakoglu** [comment]
I was actually able to reproduce an issue with the bounds and I'll be looking for a solution by tomorrow.

**Comment by bcakmakoglu** [comment]
@MoNSTRiKcom 
I can't open your video.
The issue I was able to identify yesterday actually comes from improper css around the Draggable Element.
If the containing element doesn't have any width or height set to it or a boundary that's misaligned to the displayed element then Draggable will only be able to move around in those boundaries.

Example:
```vue
<style>
.wrapper {
    display: flex;
	height: 100vh;
    width: 100vw;
}
.container {
   width: 0%;
   height: 100%;
}
</style>
<div class="wrapper">
   <div class="container">
    <Draggable bounds="body">
       <span>Drag me</span>
    </Draggable>
  </div>
</div>
```

The above example would cause the Draggable Element to only be draggable on the Y-Axis as there is no width to move in (given that the container is wrapped in another Element that provides *any* width).

```vue
<style>
.wrapper {
    display: flex;
	height: 100vh;
    width: 100vw;
}
.container {
   width: 100%;
   height: 100%;
}
</style>
<div class="wrapper">
   <div class="container">
    <Draggable bounds="body">
       <span>Drag me</span>
    </Draggable>
  </div>
</div>
```

If you instead provide with actual width and height the element becomes draggable on both axis again and is bound to the element provided in the bounds option (body in this example).

Does this relate to your issue? I was able to solve the problem I had by simply adding an actual width to the container element of my Draggable component. Now it stays properly in bounds.

**Comment by MoNSTRiKcom** [comment]
Recovered the video.
.Container has a specified width. .Wrapper doesn't. Today I will try to play with styles.

**Comment by MoNSTRiKcom** [comment]
Hi.
I checked fixed width and height - do not working.
position: absolute - does not affect 

`bounds=".sheduler-table__body"`

[Video](https://drive.google.com/file/d/1ikrzcHxudX33WIjIgOhINkhgaddMuCHY/view?usp=sharing)

The movement on the grid is still strange.

**Comment by MoNSTRiKcom** [comment]
yes, and more.
trying to dynamically affect the grid via a parameter, but the value remains static ...

**pug:**
`Draggable(:grid="[sheduler.tableWidthColumn, sheduler.tableHeightRow])"`

**Comment by bcakmakoglu** [comment]
I'll check the prop update issue.
I still can't open your videos though.

Can you give me an idea on the general HTML-Structure and Styling so I can try to recreate it?

**Comment by bcakmakoglu** [comment]
@MoNSTRiKcom 
State update issue is fixed in the next version.
I'd like to find out what your issue with the boundary is before publishing it though, if a fix is necessary.

**Comment by MoNSTRiKcom** [comment]
[Video](https://drive.google.com/file/d/1ikrzcHxudX33WIjIgOhINkhgaddMuCHY/view?usp=sharing) - update rights

Structure and styles of parent elements

```pug
#app
 #skeleton
  #main
    .main__component
      .sheduler-table
        .sheduler-table__body
          .sheduler-container
            .sheduler-container__drag // - daggable components
```

```css
#app {
    height: 100%;
}
#skeleton {
    height: 100%;
    display: grid;
    grid-gap: 1px;
    grid-template-columns: auto 1fr;
    grid-template-rows: 70px 40px auto 25px;
    grid-template-areas: "sidebar top" "sidebar tabs" "sidebar main" "sidebar status-bar";
}
#main {
    grid-area: main;
}
#main .main__component {
    height: 100%;
    position: relative;
}
.sheduler-table {
    display: grid;
    width: 100%;
    height: calc(100% - 40px);
    grid-template-columns: 60px auto;
    grid-template-rows: 40px auto;
    background: #FCFCFC;
    position: absolute;
    overflow-x: auto;
    grid-template-columns: 60px 1fr 60px;
    grid-template-rows: 40px 1fr;
    grid-template-areas: "head head head" "left body right";
}
.sheduler-table__body {
    grid-area: body;
    display: flex;
    justify-content: space-between;
    align-items: baseline;
    position: absolute;
}
.sheduler-container {
    top: 36px;
    left: 0px;
    height: 36px;
    width: 244px;
    position: absolute;
    z-index: 2;
    width: 100%;
    height: 20px;
    background-color: #27AE60;
    color: #FFF;
    border-radius: 3px;
}
.sheduler-container__drag {
    position: absolute;
    border: 1px solid #27AE60;
    border-radius: 3px;
    opacity: 0.8;
    cursor: pointer;
    box-shadow: 0px 5px 30px -2px rgba(0, 0, 0, 0.1);
    background: rgba(39, 174, 96, 0.4);
    text-align: center;
    font-weight: 600;
    color: rgba(0, 0, 0, 0.8);
}
```

**Comment by bcakmakoglu** [comment]
Thanks, I'll check it out and will get back at you as soon as I have an idea what's wrong :)

**Comment by bcakmakoglu** [comment]
I was able to replicate the issue:

Remove the `position: absolute` styling from your table-body element.
It interferes with the actual box of the boundaries.
That resolved the strange boundary box when I used your structure and stylings.
Everything else seems to work fine.
If that resolves your boundary issue I'll publish the new version which fixes the updating issue / the resetting of positions.

```css
.sheduler-table__body {
    grid-area: body;
    display: flex;
    justify-content: space-between;
    align-items: baseline;
    position: absolute; // remove this line
}
```

**Comment by MoNSTRiKcom** [comment]
Helped to remove position: absolute

at .sheduler-container and .sheduler-table__body

But the whole thing then collapses. I will think about how to apply it.

**Comment by bcakmakoglu** [comment]
@MoNSTRiKcom 
Yeah; You'll have to give the container a size and if possible not rely on absolute positions :)

I published versin 0.3.8 now, please install it and let me know if your updating / resetting positions issue is resolved!
